# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1234898642 18000
#      Tue Feb 17 14:24:02 2009 -0500
# Node ID 45524925bed9054531132c2ef45ff620435674ba
# Parent  f57c9fdb68365d0bb9a065238cf003acbdec234b
ov-class.cc (octave_class::subsref): return class object, not struct when indexing directly

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2009-02-17  John W. Eaton  <jwe@octave.org>
 
+	* ov-class.cc (octave_class::subsref): If indexing directly and
+	result is map, return class object, not simple struct.
+
 	* symtab.cc
 	(symbol_table::fcn_info::fcn_info_rep::load_private_function):
 	Pass dir_name to load_fcn_from_file.
 	
 2009-02-17  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/data.cc (NATIVE_REDUCTION): Add BOOL_FCN argument.
 	(NATIVE_REDUCTION_1): Check integer overflow flags and possibly gripe.
diff --git a/src/ov-class.cc b/src/ov-class.cc
--- a/src/ov-class.cc
+++ b/src/ov-class.cc
@@ -230,17 +230,18 @@ octave_class::subsref (const std::string
 
 		    // We handled two index elements, so tell
 		    // next_subsref to skip both of them.
 
 		    skip++;
 		  }
 	      }
 	    else
-	      retval(0) = map.index (idx.front ());
+	      retval(0) = octave_value (map.index (idx.front ()),
+					class_name ());
 	  }
 	  break;
 
 	case '.':
 	  {
 	    if (map.numel() > 0)
 	      {
 		Cell t = dotref (idx.front ());
